include/callout.h

```diff
@@ -0,0 +1,52 @@
+#ifndef __CALLOUT_H__
+#define __CALLOUT_H__
+
+#include <common.h>
+#include "queue.h"
+#include "bitset.h"
+#include "_bitset.h"
+
+typedef void (*timeout_t)(void *);
+typedef int sbintime_t;
+
+typedef struct callout {
+  TAILQ_ENTRY(callout) c_link;
+  sbintime_t c_time;  /* absolute time of the event */
+  timeout_t c_func;  /* function to call */
+  void    *c_arg;     /* function argument */
+  uint16_t c_flags;
+  unsigned int index; /* index of bucket this callout is assigned to */
```

Comment:
Please use a type that explicitly states number of bits in word, i.e. uint32_t. Please make sure you use this convention through your whole code.